asmcomp/selectgen.ml

```diff
@@ -793,9 +793,7 @@ method emit_expr (env:environment) exp =
         List.map (fun (nfail, ids, e2) ->
             let rs =
               List.map
-                (* CR-someday mshinwell: consider how we can do better than
-                   [typ_val] when appropriate. *)
-                (fun id -> let r = self#regs_for typ_val in name_regs id r; r)
+                (fun (id, typ) -> let r = self#regs_for typ in name_regs id r; r)
```

Comment:
I am not completely sure the CR should be deleted: you are
adding support for non-val parameters, but all parameters are
still passed as values. Of course, the CR make little sense here,
maybe it should just be moved *e.g.* to your addition in `Cmmgen`.

Reply:
I've now moved the CR to `Cmmgen`.